test/mocks/v1alpha1/testing_event_loop.sk.go

```diff
@@ -87,16 +87,33 @@ func (el *testingEventLoop) Run(namespaces []string, opts clients.WatchOpts) (<-
 	go errutils.AggregateErrs(opts.Ctx, errs, emitterErrs, "v1alpha1.emitter errors")
 	go func() {
 		var channelClosed bool
+
 		// create a new context for each loop, cancel it before each loop
 		var cancel context.CancelFunc = func() {}
+
 		// use closure to allow cancel function to be updated as context changes
 		defer func() { cancel() }()
+
+		// cache the previous snapshot for comparison
+		var previousSnapshot *TestingSnapshot
+
 		for {
 			select {
 			case snapshot, ok := <-watch:
 				if !ok {
 					return
 				}
+
+				if syncDecider, isDecider := el.syncer.(TestingSyncDecider); isDecider {
```

Comment:
When this is running, how should I expect to debug this behavior? Should we have metrics or logs emitted when a sync is not necessary?

Reply:
Yes, the implementation can have metrics to track this

Reply:
Is that something that we need to implement here, or in Gloo we can define that SyncDecider?